Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java

```diff
@@ -323,4 +365,182 @@ private static ListCertificatesResponseMessageType buildListCertificatesResponse
+                ALGORITHM_SHA256_RSA);
+            KeyStore jksGatewayNew = getGatewayNew();
+            if (jksGatewayNew.containsAlias(alias)) {
+                jksGatewayNew.deleteEntry(alias);
+            }
+            jksGatewayNew.setKeyEntry(alias, pair.getPrivate(),
+                System.getProperty(CertificateManagerImpl.KEY_STORE_PASSWORD_KEY).toCharArray(),
+                CoreHelpUtils.getCertificateChain(x509cert));
+            CoreHelpUtils.saveJksTo(jksGatewayNew, System.getProperty(CertificateManagerImpl.KEY_STORE_PASSWORD_KEY),
+                FILE_JKS_GATEWAY_NEW);
+        } catch (Exception e) {
+            LOG.error("error while generating the certificate for gateway_new.jks: {}", alias, e);
+            return buildSimpleResponse(false, "fail to generate the certificate: " + alias);
+        }
+        return buildSimpleResponse(true, ACT_SUCCESSFUL);
+
+    }
+
+    private static SimpleCertificateResponseMessageType buildSimpleResponse(boolean status, String msg) {
+        SimpleCertificateResponseMessageType resp = new SimpleCertificateResponseMessageType();
+        resp.setStatus(status);
+        resp.setMessage(msg);
+        return resp;
+    }
+
+    private static KeyStore getGatewayNew() throws CertificateManagerException {
+        if (null == gatewayNew) {
+            File destinationFile = new File(FILE_JKS_GATEWAY_NEW);
+            if(!destinationFile.exists()){
+                copyFile(FILE_JKS_GATEWAY, FILE_JKS_GATEWAY_NEW);
+            }
+            gatewayNew = CertificateUtil.loadKeyStore(CertificateManagerImpl.JKS_TYPE,
+                System.getProperty(TRUST_STORE_PASSWORD_KEY), FILE_JKS_GATEWAY_NEW);
+        }
+        return gatewayNew;
+    }
+
+    private static int getKeySize() {
+        try{
+            return Integer
+                .parseInt(prop.getProperty(CA_AUTHORITY_PROPERTY_FILE, KEYSIZE, String.valueOf(DEFAULT_KEYSIZE)));
+        }catch (NumberFormatException e) {
+            LOG.debug("unable to parse keysize: {}, {}", CA_AUTHORITY_PROPERTY_FILE, KEYSIZE, e);
+            return DEFAULT_KEYSIZE;
+        }
+    }
+
+    private static int getValidityDays() {
+        try {
+            return Integer
+                .parseInt(prop.getProperty(CA_AUTHORITY_PROPERTY_FILE, VALIDITY, String.valueOf(DEFAULT_VALIDITY)));
+        } catch (NumberFormatException e) {
+            LOG.debug("unable to parse keysize: {}, {}", CA_AUTHORITY_PROPERTY_FILE, VALIDITY, e);
+            return DEFAULT_VALIDITY;
+        }
+    }
+
+    private static boolean copyFile(String sourceFileName, String destinationFileName) {
+
+        File sourceFile = new File(sourceFileName);
+        File destinationFile = new File(destinationFileName);
+
+        LOG.debug("copy file from-to: {}, {}", sourceFileName, destinationFileName);
+
+        if (sourceFile.exists()) {
+            try {
+                org.apache.commons.io.FileUtils.copyFile(sourceFile, destinationFile, false);
```

Comment:
Why do we have the full name here? I did not see another FileUtils being used here.